### PR TITLE
importbot importing books that are of bad quality

### LINKED ISSUE #4151: Fix BWB Importbot Low Quality records
Importbot is importing low-quality records from unspecified source. Titles are public domain reprints (possibly print on demand) with keyword-stuffed titles. 

### Evidence / Screenshot (if possible)
Screenshot 2020-11-24 at 21 31 56

Example: https://openlibrary.org/search?q=adventures+of+sherlock+holmes+conan+doyle&mode=everything&page=3

### Proposal & Constraints

Books have valid ISBNs and human editors should be allowed to add manually but these should be blocked from import by importbot.

### Stakeholders

@mekarpeles

### PR DESCRIPTION
**Pull request by jennshan**
in this commit a new function is added that checks a book's title and other attributes for known exclusions.
The effect of this commit is that, the importbot will discriminate low quality books

Closes #4151 

hotfix

### Technical

This is not a complete solution yet, it is just an initial try

### Testing

The importbot must be its periodic imports, filtering books that have the keyword "Notebook" in their titles and the publisher is independent publisher.

### CODE REVIEW

File: scripts/partner_batch_imports.py [hunk 1]
```diff
@@ -173,6 +173,14 @@ def csv_to_ol_json_item(line):
     b = Biblio(data)
     return {'ia_id': b.source_id, 'data': b.json()}
 
+def is_low_quality_book(book_item):
+    """check if a book item is of low quality"""
+    book_title_breakup = book_item.title.split()
+    if "Notebook" in book_title_breakup and book_item.publisher == "Independently Published":
```
Comment (jimman2003) [inline scripts/partner_batch_imports.py:179]: You should also check for "notebook" in title , solr is case-insensitive, I believe

**Review by jimman2003 — CHANGES_REQUESTED** [review]
You should also check for "notebook" in title , solr is case-insensitive, I believe

File: scripts/partner_batch_imports.py (hunk 1 above)
Comment (jimman2003) [inline scripts/partner_batch_imports.py:179]: You should test the substring in a case-insensitive way,as I believe solr also does
```suggestion
    if "notebook" in book_title.casefold() and "independently published" in book_item.publisher.casefold():
```

File: scripts/partner_batch_imports.py [hunk 2]
```diff
@@ -173,6 +173,14 @@ def csv_to_ol_json_item(line):
     b = Biblio(data)
     return {'ia_id': b.source_id, 'data': b.json()}
 
+def is_low_quality_book(book_item):
+    """check if a book item is of low quality"""
+    book_title_breakup = book_item.title.split()
```
Comment (jimman2003) [inline scripts/partner_batch_imports.py:178]: No need to create a list to test a substring 
```suggestion
    
```

**Review by jimman2003 — CHANGES_REQUESTED** [review]
This function can be rewritten even more readable as ```
return (notebook" in book_title.casefold() and "independently published" in book_item.publisher.casefold())```. Alternatively:

**Comment by mekarpeles** [comment]
Thank you @jennshan for taking a stab at this and @jimman2003 for your review

This isn't the easiest issue to get started with and we really appreciate your contribution to fixing a big set of problems regarding imports!

File: scripts/partner_batch_imports.py (hunk 1 above)
Comment (jennshan) [reply scripts/partner_batch_imports.py]: changed the code, now the test is case-insensitive. I assumed you meant book_item.title (book_item is the argument to the function is_low_quality_book)

**Comment by mekarpeles** [comment]
Looks like a good first step! Nicely done @jennshan! And thank you for helping lead the code review @jimman2003!

File: scripts/partner_batch_imports.py [hunk 3]
```diff
@@ -173,6 +173,10 @@ def csv_to_ol_json_item(line):
     b = Biblio(data)
     return {'ia_id': b.source_id, 'data': b.json()}
 
+def is_low_quality_book(book_item):
+    """check if a book item is of low quality"""
+    return ("notebook" in book_item.title.casefold() and "independently published" in book_item.publisher.casefold())
```
Comment (tfmorris) [inline scripts/partner_batch_imports.py:178]: This appears to be English-only code. Is that still allowed?

**Comment by LeadSongDog** [comment]
Glad to see some restraint is being applied toImportBot, but it’s a much bigger problem than just notebooks. A few bad actors are creating massive numbers of bogus editions of classic works (many taken *from* IA), listing them on BWB, and we are re-inhaling them willy-nilly with dozens or even hundreds of ISBNs each. Many have BWB listings with bogus extra authors such as "Mint Editions": these lead to extra work records too!

7A1610DF-E15C-4AB4-8B52-5FD6697AD13E

File: scripts/partner_batch_imports.py (hunk 3 above)
Comment (cdrini) [reply scripts/partner_batch_imports.py:178]: The bulk of the spammy books were in English; we should add more checks here as we find more bad auto-imports. ( https://docs.google.com/spreadsheets/d/1mG8Tn-sx73fdcHNAvFCJ-2p_-ULT_5QH90y5LLMpbtY/edit#gid=0 ). Good choice of query, @jennshan !

**Pull request merged**